File: Week 2/Week 2_ Templates/C/C_freeweights.cpp
#include <bits/stdc++.h> 
using namespace std;

int free_weights(int N, vector<int>& a, vector<int>& b){
    // Your code to find heaviest weight required goes here
    return -1;
}

int main(){
    int n;
    cin >> n;
    vector<int> a(n), b(n);
    for(int i = 0; i < n; i++){
        cin >> a[i];
    }
    for(int i = 0; i < n; i++){
        cin >> b[i];
    }
    cout << free_weights(n, a, b) << endl;
}   

File: Week 2/Week 2_ Templates/C/C_freeweights.java
import java.io.*;
import java.util.*;

// make sure you submit w/ the class name set to Main if you use this template
class Main{
    static int freeWeights(int n, ArrayList<Integer> a, ArrayList<Integer> b){
        // Your code to find the minimum weight required goes here
        return -1;
    }
    public static void main(String[] args) throws IOException{
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(in.readLine());
        ArrayList<Integer> a = new ArrayList<Integer>();
        ArrayList<Integer> b = new ArrayList<Integer>();
        
        StringTokenizer st = new StringTokenizer(in.readLine());
        for(int i = 0; i < n; i++){
            a.add(Integer.parseInt(st.nextToken()));
        }
        st = new StringTokenizer(in.readLine());
        for(int i = 0; i < n; i++){
            b.add(Integer.parseInt(st.nextToken()));
        }
        
        System.out.println(freeWeights(n, a, b));
    }
}